File: japiwznm/WznmQTcoList.java
/**
  * \file WznmQTcoList.h
  * Java API code for record of table TblWznmQTcoList
  * \author Alexander Wirthmueller
  * \date created: 27 Aug 2020
  * \date modified: 27 Aug 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQTcoList {
	
	public WznmQTcoList(
				int jnum
				, String sref
				, String Short
				, String srefIxVBasetype
				, String titIxVBasetype
				, String stubTblRefWznmMTable
				, String srefFctIxVTbl
				, String titFctIxVTbl
				, String srefIxVSubtype
				, String titIxVSubtype
				, String srefIxVAxisfct
				, String titIxVAxisfct
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.Short = Short;
		this.srefIxVBasetype = srefIxVBasetype;
		this.titIxVBasetype = titIxVBasetype;
		this.stubTblRefWznmMTable = stubTblRefWznmMTable;
		this.srefFctIxVTbl = srefFctIxVTbl;
		this.titFctIxVTbl = titFctIxVTbl;
		this.srefIxVSubtype = srefIxVSubtype;
		this.titIxVSubtype = titIxVSubtype;
		this.srefIxVAxisfct = srefIxVAxisfct;
		this.titIxVAxisfct = titIxVAxisfct;
	};

	public int jnum;
	public String sref;
	public String Short;
	public String srefIxVBasetype;
	public String titIxVBasetype;
	public String stubTblRefWznmMTable;
	public String srefFctIxVTbl;
	public String titFctIxVTbl;
	public String srefIxVSubtype;
	public String titIxVSubtype;
	public String srefIxVAxisfct;
	public String titIxVAxisfct;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQTcoList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			Short = Xmlio.extractStringUclc(doc, basexpath, "Short", "sho", null, 0);
			srefIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVBasetype", "typ", null, 0);
			titIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "titIxVBasetype", "typ2", null, 0);
			stubTblRefWznmMTable = Xmlio.extractStringUclc(doc, basexpath, "stubTblRefWznmMTable", "tbl", null, 0);
			srefFctIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "srefFctIxVTbl", "fct", null, 0);
			titFctIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "titFctIxVTbl", "fct2", null, 0);
			srefIxVSubtype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVSubtype", "sty", null, 0);
			titIxVSubtype = Xmlio.extractStringUclc(doc, basexpath, "titIxVSubtype", "sty2", null, 0);
			srefIxVAxisfct = Xmlio.extractStringUclc(doc, basexpath, "srefIxVAxisfct", "axf", null, 0);
			titIxVAxisfct = Xmlio.extractStringUclc(doc, basexpath, "titIxVAxisfct", "axf2", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQTcoList.cpp
/**
	* \file WznmQTcoList.cpp
	* API code for table TblWznmQTcoList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WznmQTcoList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQTcoList
 ******************************************************************************/

WznmQTcoList::WznmQTcoList(
			const uint jnum
			, const string sref
			, const string Short
			, const string srefIxVBasetype
			, const string titIxVBasetype
			, const string stubTblRefWznmMTable
			, const string srefFctIxVTbl
			, const string titFctIxVTbl
			, const string srefIxVSubtype
			, const string titIxVSubtype
			, const string srefIxVAxisfct
			, const string titIxVAxisfct
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->Short = Short;
	this->srefIxVBasetype = srefIxVBasetype;
	this->titIxVBasetype = titIxVBasetype;
	this->stubTblRefWznmMTable = stubTblRefWznmMTable;
	this->srefFctIxVTbl = srefFctIxVTbl;
	this->titFctIxVTbl = titFctIxVTbl;
	this->srefIxVSubtype = srefIxVSubtype;
	this->titIxVSubtype = titIxVSubtype;
	this->srefIxVAxisfct = srefIxVAxisfct;
	this->titIxVAxisfct = titIxVAxisfct;
};

bool WznmQTcoList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQTcoList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "Short", "sho", Short);
		extractStringUclc(docctx, basexpath, "srefIxVBasetype", "typ", srefIxVBasetype);
		extractStringUclc(docctx, basexpath, "titIxVBasetype", "typ2", titIxVBasetype);
		extractStringUclc(docctx, basexpath, "stubTblRefWznmMTable", "tbl", stubTblRefWznmMTable);
		extractStringUclc(docctx, basexpath, "srefFctIxVTbl", "fct", srefFctIxVTbl);
		extractStringUclc(docctx, basexpath, "titFctIxVTbl", "fct2", titFctIxVTbl);
		extractStringUclc(docctx, basexpath, "srefIxVSubtype", "sty", srefIxVSubtype);
		extractStringUclc(docctx, basexpath, "titIxVSubtype", "sty2", titIxVSubtype);
		extractStringUclc(docctx, basexpath, "srefIxVAxisfct", "axf", srefIxVAxisfct);
		extractStringUclc(docctx, basexpath, "titIxVAxisfct", "axf2", titIxVAxisfct);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQTcoList
 ******************************************************************************/

ListWznmQTcoList::ListWznmQTcoList() {
};

ListWznmQTcoList::ListWznmQTcoList(
			const ListWznmQTcoList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQTcoList(*(src.nodes[i]));
};

ListWznmQTcoList& ListWznmQTcoList::operator=(
			const ListWznmQTcoList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQTcoList(*(src.nodes[i]));

	return *this;
};

ListWznmQTcoList::~ListWznmQTcoList() {
	clear();
};

void ListWznmQTcoList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQTcoList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQTcoList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQTcoList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQTcoList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQTcoList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQTcoList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

